include/fixed_containers/fixed_red_black_tree_view.hpp

```diff
@@ -0,0 +1,386 @@
+            const auto storage_ptr = bptr;
+            const auto fixed_vector_ptr = storage_ptr;
+            return *reinterpret_cast<const std::size_t*>(fixed_vector_ptr);
+        }
+
+        /**
+         * Calculate the size of each element in the storage's internal buffer.
+         */
+        constexpr std::size_t storage_elem_size_bytes() const
+        {
+            const auto node_size_bytes = tree_node_size_bytes();
+
+            switch (storage_type_)
+            {
+            case StorageType::FIXED_INDEX_POOL:
+                // IndexOrValueStorage is a union containing a size_t (index) or the node itself.
+                return std::max(sizeof(std::size_t), node_size_bytes);
+
+            case StorageType::FIXED_INDEX_CONTIGUOUS:
+                return node_size_bytes;
+            }
+
+            assert(false && "unreachable");
+        }
+        
+        /**
+         * Calculate the size of each tree node used in the red-black tree, using the input sizes
+         * as the size of the key and value types.
+         */
+        std::size_t tree_node_size_bytes() const
+        {
+            using namespace fixed_red_black_tree_detail;
+
+            /* 
+             * Calculate the size of the tree node by calculating the compile-time size with a
+             * native pointer type as the Key, adding the size of the value type, subtracting the
+             * pointer size from the result and rounding up for alignment.
+             */
+            std::size_t base_node_size_bytes;
+            switch (compactness_.backing_enum())
+            {
+            case CompactnessBacking::DEDICATED_COLOR:
+                base_node_size_bytes = sizeof(DefaultRedBlackTreeNode<uintptr_t>);
+                break;
+            case CompactnessBacking::EMBEDDED_COLOR:
+                base_node_size_bytes = sizeof(CompactRedBlackTreeNode<uintptr_t>);
+                break;
+            }
+
+            return align_up(
+                base_node_size_bytes - sizeof(uintptr_t) + elem_size_bytes_,
+                sizeof(uintptr_t));
+        }
+
+    private:
+        const std::byte* base_;
+        std::size_t elem_size_bytes_;
+        std::size_t max_size_bytes_;
+        Compactness compactness_;
+        StorageType storage_type_;
+        std::size_t storage_elem_size_bytes_;
+        
+        NodeIndex index_;
+        const std::byte* cur_pointer_;
+    };
+
+public:
+    FixedRedBlackTreeRawView(const void* tree_ptr,
+                             std::size_t elem_size_bytes,
+                             std::size_t max_size_bytes,
+                             Compactness compactness,
+                             StorageType storage_type)
+      : tree_ptr_(reinterpret_cast<const std::byte*>(tree_ptr))
+      , elem_size_bytes_(elem_size_bytes)
+      , max_size_bytes_(max_size_bytes)
+      , compactness_(compactness)
+      , storage_type_(storage_type)
```

Comment:
Nit: Consider `{}` for initialization instead of `()`.
Differences are explained here: https://www.reddit.com/r/cpp_questions/comments/f3dcs7/comment/fhk8ruf/

Reply:
05aa046